pkg/testutil/testutil.go

```diff
@@ -0,0 +1,85 @@
+package testutil
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/gitpod-io/leeway/pkg/leeway"
+	"gopkg.in/yaml.v3"
+)
+
+type Setup struct {
+	Workspace  leeway.Workspace `yaml:"workspace"`
+	Components []Component      `yaml:"components"`
+}
+
+type Component struct {
+	Location string            `yaml:"location"`
+	Files    map[string]string `yaml:"files"`
+	Comp     leeway.Component  `yaml:"comp"`
+	Packages []leeway.Package  `yaml:"packages"`
+	Scripts  []leeway.Script   `yaml:"scripts"`
+}
+
+// LoadFromYAML loads a workspace setup from a YAML file
+func LoadFromYAML(in io.Reader) (*Setup, error) {
+	fc, err := io.ReadAll(in)
+	if err != nil {
+		return nil, err
+	}
+
+	var res Setup
+	err = yaml.Unmarshal(fc, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	return &res, nil
+}
+
+// Materialize produces a leeway workspace according to the setup spec
+func (s Setup) Materialize() (workspaceRoot string, err error) {
+	workspaceRoot, err = os.MkdirTemp("", "leeway-test-*")
+	if err != nil {
+		return
```

Comment:
Is it on purpose that you don't return the error in this all the other `err != nil` cases in this function? ☺️

Reply:
`err` is a named return parameter. `return` will hence return the value of `err` set in line 43

Reply:
Ah! That makes sense, thanks!